### PR TITLE
Adding WebSocket IO for Alex.

### PR DESCRIPTION
**Pull request by ticcky**
WebSocket IO for Alex consists of:
  - WSHub - which is a Hub that runs Alex with WebSocket as it's input/output protocol (instead of for example VOIP)
  - WSRouter - which is a program that manages running Alex instances and routes clients to a free Alex instance.

### CODE REVIEW

File: alex/applications/wshub.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
+#!/usr/bin/env python
+"""
```
Comment (tuetschek) [inline alex/applications/wshub.py]: Is this here on purpose?

Comment (ticcky) [inline alex/applications/wshub.py]: Good catch, thanks! =)

**Comment by tuetschek** [comment]
I have tested this branch using `vhub_test_kaldi_hdc_slu` and it worked correctly.

The only thing is that I am getting a lot of log messages on the console that look like this (one per 3 seconds):

    EXEC Time inner loop: DM t = 1.0010 c = 0.0016

@ticcky: Are these log messages specific to your changes, or is it a coincidence that I did not get them on master but did get them with your branch? Do we need them for something?

@jurcicek : Do you want me to test a different configuration as well?

**Comment by jurcicek** [comment]
no, it should be good enough

**Comment by ticcky** [comment]
Thanks Ondra, I fixed the printouts. Their intention, I believe, is to print out a message if the execution is slow.

**Comment by tuetschek** [comment]
Thanks – I tested it again with `vhub_test_kaldi_hdc_slu` and everything worked OK.

**Pull request merged**